ref(models): add data forwarding

liuirene256 opened the pull request:
Data Forwarding Models. `data_forwarding_config.py` has the general config model (per org) and `data_forwarding_project.py` is the config for each project (including project level config overrides).

Commit:
add data forwarding

leeandher commented on src/sentry/integrations/models/data_forwarding_config.py:
```diff
@@ -0,0 +1,29 @@
+    is_enabled = models.BooleanField(default=True)
+
+    enroll_new_projects = models.BooleanField(default=False)
+    enrolled_projects = JSONField(default=list)
```
models.JSONField is preferred, so lets switch to that one! (e.g. https://github.com/getsentry/sentry/pull/99209)

leeandher commented on src/sentry/integrations/models/data_forwarding_config.py:
```diff
@@ -0,0 +1,29 @@
+
+    __relocation_scope__ = RelocationScope.Organization
+
+    organization_id = FlexibleForeignKey("sentry.Organization", on_delete=models.CASCADE)
```
Good call with the cascade deletion!

leeandher commented on src/sentry/integrations/models/data_forwarding_config.py:
```diff
@@ -0,0 +1,29 @@
+    enroll_new_projects = models.BooleanField(default=False)
+    enrolled_projects = JSONField(default=list)
+
+    provider = models.CharField(max_length=64)
```
I know the `Integration` model doesn't do it, but you could add some choices here, it's done in the docs https://docs.djangoproject.com/en/5.2/ref/models/fields/#choices. Mostly an FYI though, not sure it's necessary to change.

leeandher commented on src/sentry/integrations/models/data_forwarding_config.py:
```diff
@@ -0,0 +1,29 @@
+
+
+@region_silo_model
+class DataForwardingConfig(DefaultFieldsModel):
```
I probably set you up for this (mb) but I think the name data forwarding config should be something else, only because

```python
config = DataForwardingConfig.objects.get(...)
property = config.config.get(...) # this is kinda confusing
```

Maybe `DataForwardingIntegration` or `DataForwarder`?

Or you can change the actual field from `config` to `spec` or `setting`, just so they don't conflict.

leeandher commented on src/sentry/integrations/models/data_forwarding_config.py:
```diff
@@ -0,0 +1,29 @@
+@region_silo_model
+class DataForwardingConfig(DefaultFieldsModel):
+    """
+    Configuration for data forwarding to external services (Segment, SQS, or Splunk).
+    This is a general config that can apply to specific projects or all projects in an organization.
+    """
```
🔥  great description!

leeandher commented on src/sentry/integrations/models/data_forwarding_project.py:
```diff
@@ -0,0 +1,40 @@
+    config = FlexibleForeignKey(
+        "sentry.DataForwardingConfig",
+        on_delete=models.CASCADE,
+        related_name="project_configs",
```
I think `projects` makes sense as a shorter related name, unless it can't be used for some reason.

leeandher commented on src/sentry/integrations/models/data_forwarding_project.py:
```diff
@@ -0,0 +1,40 @@
+        related_name="project_configs",
+    )
+    project_id = FlexibleForeignKey("sentry.Project", on_delete=models.CASCADE)
+    overrides = JSONField(default=dict)
```
same note as above about `models.JSONFIeld`, you'll have to regenerate the migration btw.

leeandher reviewed (commented):
Some of my comments may require reissuing the migration, we have some notes on this in the docs https://develop.sentry.dev/backend/application-domains/database-migrations/#automatically-resolving-conflicts

In your devenv

```
$ bin/update-migration <migration_name>
```

getsantry[bot] committed:
:hammer_and_wrench: apply pre-commit fixes

Commit:
switch to models.JSONField

Commit:
regenerate migration

Commit:
fix tests

Commit:
regen

Commit:
change metadata name

liuirene256 committed:
Merge branch 'master' into irene/models

michellewzhang committed:
ref(releases): fix issue count on release index & other spots (#99555)

closes
https://linear.app/getsentry/issue/REPLAY-706/release-newgroups-count-is-sometimes-incorrect
closes
https://linear.app/getsentry/issue/REPLAY-151/release-overview-new-issues-counter-doesnt-filter-when-selecting
closes
https://linear.app/getsentry/issue/REPLAY-139/new-issues-in-release-showing-up-unexpectedly-in-the-team-plan

i believe all the issues above were relating to an incorrect issue count
value in the release row, leading to confusion about the actual count of
new issues for that release.

the `newGroups` value returned by the `/releases/` endpoint is correct;
however, the frontend on the releases index was mistakenly using the
project's `newGroups` field.  The project's `newGroups` field is the sum
of all the new issues for that project -- hence why they're all the same
in the photo below. we should instead be using the count from the
release itself.

i also fixed 2 other spots where we were relying on the project
`newGroups` field in insights mobile session health page & on release
drawer.

before:
<img width="1084" height="376" alt="SCR-20250915-nxra"
src="https://github.com/user-attachments/assets/2f2044b7-4cb2-49e1-b2fb-437589bf0212"
/>

<img width="1272" height="514" alt="SCR-20250915-oech"
src="https://github.com/user-attachments/assets/d6b7d321-c91b-4221-a673-bdc2656c40a5"
/>


after:
<img width="1080" height="346" alt="SCR-20250915-nxnn"
src="https://github.com/user-attachments/assets/a72f03fd-e478-4e22-a6c8-217958674cfc"
/>

<img width="1267" height="564" alt="SCR-20250915-ocyn"
src="https://github.com/user-attachments/assets/65364c90-f9de-459a-af16-82ab1313fdd8"
/>

markstory committed:
fix(tasks) Change usage metric tag (#99585)

Using `app_feature` results in duplicate tags in saas, as our metrics
infrastructure also applies an `app_feature` tag as well.

Refs STREAM-445

roggenkemper committed:
chore(detectors): Update experimental N+1 API detector to use rollout flag (#99556)

adds flag so we can roll out experimental n+1 api call detector.

Commit:
fix test

shruthilayaj committed:
feat(explore): Allow boolean types in count_if (#99591)

Ahmed-Labs committed:
chore(dashboards): Register dashboard global filters feature flag (#99582)

Registers a feature flag to gate the development of dashboard global
filters.

evanpurkhiser committed:
ref(ts): Mark some react router types as deprecated (#99580)

narsaynorath committed:
fix(dashboards): has filter only shows aggregates once they are selected (#99579)

The filter bar was showing aggregates even though no aggregates were
selected, so you could do something like `has:p50` but that is
non-sensical.

The behaviour in Discover is that these fields should only appear once
they're selected in the query. e.g. imagine on Discover you had a table
aggregating on transaction name and p50(transaction.duration), you're
able to filter out rows by the p50 value. If it's not there, it should
not be an option shown.

I also filtered functions out of the `has:` results because `has:` is
for event attributes and not for aggregates. The tests for both
components are _slightly_ similar because they do similar things.

malwilley committed:
chore(nav-cleanup): Clean up nav references in view components (#99529)

Commit:
fix test

michellewzhang committed:
fix(releases): fix error count dataset query (#99596)

the errors chart in release details was incorrectly querying the
`metricsEnhanced` dataset instead of the errors dataset.

closes
https://linear.app/getsentry/issue/REPLAY-433/release-page-doesnt-render-graph-but-clearly-has-data


<img width="719" height="317" alt="SCR-20250916-jyfk"
src="https://github.com/user-attachments/assets/a1b150ca-a4f2-4b22-b534-92f281b6fdd6"
/>

before:

<img width="908" height="709" alt="SCR-20250916-kabo"
src="https://github.com/user-attachments/assets/c0677b3a-cf67-4662-ab0c-0599b105c2fe"
/>

after:

<img width="998" height="703" alt="SCR-20250916-kacp"
src="https://github.com/user-attachments/assets/86f84090-beb5-4eec-8cd3-afcbc3651ffe"
/>

cvxluo committed:
ref(issues): use `RateLimitConfig` instead of deprecated dict (#99490)

See https://github.com/getsentry/sentry/pull/99484, removing the
deprecated rate limit config so we can move to only using
`RateLimitConfig`.

Commit:
fix test
